lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -363,6 +366,7 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             OffsetIndex offsetIndex = getFilteredOffsetIndex(currentRowGroup, currentBlockMetadata.getRowCount(), metadata.getPath());
             List<Slice> slices = allocateBlock(fieldId);
             columnReader.setPageReader(createPageReader(slices, metadata, columnDescriptor, offsetIndex), currentGroupRowRanges);
+            codecsUsed.add(metadata.getCodec().name());
```

Comment:
For each compression codec we should record the size of data seen with that codec so that we get some idea about the amount of data with each type of codec (add up the disk range lengths).
That way the value part of the metric also becomes something useful.

Reply:
For Orc as well we can get the size of data from the StripeInformation in Footer in OrcReader